### PR TITLE
FLUID-6624: Update migration docs for Infusion 2.0 to 3.0

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-6624

### CODE REVIEW

**Comment by jobara** [comment]
@amb26 could you please review this? Is there anything else we should document? Is there anything that needs to be deprecated?

**Comment by amb26** [comment]
I had thought that the "Nexus API methods" at https://docs.fluidproject.org/infusion/development/iocapi#nexus-api-methods were new in 3.0 but amazingly they do actually seem to have been in Infusion 2.0 but were not advertised in the 1.5 to 2.0 API updates. In fact the entirety of the IoC API listed on the page looks like it was new in that release.

We should probably deprecate fluid.table and fluid.pagedTable which will be removed in Infusion 4 or 5 (although we will keep fluid.pager)

**Comment by jobara** [comment]
> We should probably deprecate fluid.table and fluid.pagedTable which will be removed in Infusion 4 or 5 (although we will keep fluid.pager)

Thanks, I've added a page about deprecations that just lists these two.

File: src/documents/APIChangesFrom2_0To3_0.md [hunk 1]
```diff
@@ -19,6 +19,11 @@ This section describes major APIs that were in common use. For information about
 * `fluid.transforms.valueMapper` takes an `defaultInput` option to provide the model data directly. This also provides a
    location for adding nested transformations.
 
+#### Fluid Document
+
+`fluid.focus` and `fluid.blur` now return a ES6 promises resolved when the related DOM event fires. In Infusion 2.0,
```
Comment (amb26) [inline src/documents/APIChangesFrom2_0To3_0.md:24]: typo promises -> promise

File: src/documents/APIChangesFrom2_0To3_0.md [hunk 2]
```diff
@@ -99,16 +104,152 @@ keyword has been changed to `"value"`.
 }
 ```
 
+#### Preference Editor
+
+##### Selectors
+
+The `panels` selector, by default `.flc-prefsEditor-panel`, is used to find all panel containers within the preferce
```
Comment (amb26) [inline src/documents/APIChangesFrom2_0To3_0.md:111]: Typo preferce -> preference

File: src/documents/APIChangesFrom2_0To3_0.md [hunk 3]
```diff
@@ -99,16 +104,152 @@ keyword has been changed to `"value"`.
...
+            </button>
+        </span>
+    </div>
+
+    <!-- This is the div that will contain the Preference Editor component -->
+    <div class="flc-slidingPanel-panel flc-prefsEditor-iframe"></div>
+
+    <!--
+        This div is for the sliding panel bar that shows and hides the Preference Editor controls in the desktop view.
+        A separate panel bar for desktop displays is needed to preserver the correct tab order.
```
Comment (amb26) [inline src/documents/APIChangesFrom2_0To3_0.md:163]: Typo preserver -> preserve

File: src/documents/DeprecatedIn3_0.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+title: Deprecated in 3.0
+category: Infusion
+---
+
+This page contains a list of the features, APIs, and etc. that are deprecated as of Infusion 3.0.
+
+<div class="infusion-docs-note"><strong>Note:</strong> While the intention is to provide advance notice of future
+changes, the contents of this page may not be exhaustive.</div>
+
+## Grades
```
Comment (amb26) [inline src/documents/DeprecatedIn3_0.md:11]: I realise we didn't get around to deprecating the old renderer in 2.0

We should add to this list everything in the fluid.renderer.* namespace as well as fluid.render, fluid.renderTemplates, fluid.reRender, fluid.selfRender - in fact everything in the files fluidRenderer.js, RendererUtilities.js, fluidParser.js as well as everything provided in fastXmlPull.js, that is, fluid.XMLP.* to the extent anyone was ever aware of that

**Comment by jobara** [comment]
@amb26 I've fixed the typos and updated the deprecations. Ready for more review.

**Pull request merged**